.rulesync/rules/playwright-locators.md

```diff
@@ -0,0 +1,115 @@
+---
+targets:
+  - '*'
+root: false
+description: Best practices for using Playwright locators in RHDH e2e tests
+globs:
+  - e2e-tests/**
+---
+
+# Playwright Locator Best Practices
+
+## Locator Priority (Use in Order)
+
+1. **`page.getByRole(role, { name })`** - Interactive elements, headings (reflects user perception)
+2. **`page.getByLabel(text)`** - Form controls with labels
+3. **`page.getByPlaceholder(text)`** - Inputs without labels
+4. **`page.getByText(text)`** - Non-interactive content only
```

Comment:
nitpick: Avoid getByText for buttons or links when a proper role-based locator exists.

Avoid targeting text that may change due to backend data (e.g., dynamic status, timestamps).

Prefer toHaveText / toContainText in assertions because they include auto-waiting, which makes tests more stable.